arch/arm/src/sama5/sam_tsd.c

```diff
@@ -245,12 +257,7 @@ static const struct file_operations g_tsdops =
 
 /* The driver state structure is pre-allocated. */
 
-static struct sam_tsd_s g_tsd =
-{
-  .threshx = INVALID_THRESHOLD,
-  .threshy = INVALID_THRESHOLD,
-  .waitsem = SEM_INITIALIZER(0),
-};
+static struct sam_tsd_s g_tsd;
```

Comment:
why this is changed?

Reply:
No idea...it seems my changes are conflicting with other changes so either my rebase/merge/whatever its called, to master  at that time, or later, who knows didn't do what I thought. This is SOOOO hard for me but really trying to make this code work so others can use it if they ever want to as it never would have worked at all in the real world.

I will give up on this PR, do may changes again, and make a clean one.